lib/linked_list.rb

```diff
@@ -1,45 +1,104 @@
-require_relative 'node'
+require_relative "node"
 
 class LinkedList
   attr_reader :head
 
+  # *Notes:
+  # when .next_node is nil current node is the last element
+
   def initialize
+    # If the head is nil it means the list is empty!
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1) --> It doen not travese the the list to add the first value
+  # Space complexity - O(1)
   def add_first(data)
+    # Create a new node
+    new_node = Node.new(data)
+    if @head.nil?
+      @head = new_node
+      return
+    end
+
+    new_node.next = @head 
+    # conection between the new node and the next one that is the head.
+    @head = new_node # The new node becomes the head.
```

Comment:
This is good, but it can be simplified a bit.

```suggestion
    # remember Node#initialize has a 2nd argument which is the new node's next field
    @head = Node.new(data, @head)
```